src/main.rs

```diff
@@ -21,25 +22,50 @@ mod joypad;
 use vm::{Vm, WIDTH, HEIGHT};
 use joypad::{JoypadKey};
 
+const MAX_ENLARGE_SCALE: usize = 5;
+
+fn arg_check_range<T>(arg: &str, range: (T, T)) -> Result<T, String>
+    where T: Ord + std::str::FromStr + std::fmt::Display
+{
+    let (min, max) = range;
+    match arg.parse::<T>() {
+        Ok(n) if min <= n && n <= max => Ok(n),
```

Comment:
Please note that your implementation will let 0 pass the test. Either change <= to < or pass 1 as min value.

Reply:
I wanted to use 1 as min. I am sorry about making this mistake.

Reply:
No need to sorry about this. People make mistake, that is why there is an eraser on every pencil.

Reply:
Thanks!